card-validation/src/components/CardForm/index.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from 'react';
+import { Formik, Form, Field } from 'formik';
+import * as Yup from 'yup';
+import Card from '../Card';
+import './index.scss';
+
+const ValidationSchema = Yup.object().shape({
+  name: Yup.string()
+    .matches(/^((?:[A-Za-z]+ ?){1,3})$/, 'enter a valid name')
+    .required('required'),
+  cardnumber: Yup.string()
+    .matches(/^4[0-9]{12}(?:[0-9]{3})?$/, 'enter a valid card number')
+    .required('required'),
+  expireMM: Yup.string().required('required'),
+  expireYY: Yup.string().required('required'),
+  securitycode: Yup.string()
+    .min(3, 'too short')
+    .max(4, 'too long')
+    .required('required'),
+});
+
+export default function CardForm({ onSubmit }) {
+  const [side, setSide] = useState('front');
+
+  const handleSubmit = (values) => {
+    console.log('values', values);
+    onSubmit(values);
+  };
+
+  return (
+    <Formik
+      initialValues={{
+        name: '',
+        cardnumber: '',
+        expireMM: '',
+        expireYY: '',
+        securitycode: '',
+      }}
+      validationSchema={ValidationSchema}
+      onSubmit={handleSubmit}
+      data-testid="form"
+    >
+      {({ errors, touched, values }) => (
+        <Form className="form">
+          <Card values={values} side={side} />
+          <div className="field">
+            <label htmlFor="name" className="field__label">
+              Name
+            </label>
+            <Field
+              name="name"
+              aria-label="name"
+              type="text"
+              className="field__input"
+              onClick={() => setSide('front')}
+            />
+            {errors.name && touched.name ? (
+              <span className="field__error">{errors.name}</span>
+            ) : null}
+          </div>
+          <div className="field">
+            <label htmlFor="cardnumber" className="field__label">
+              Card Number
+            </label>
+            <Field
+              name="cardnumber"
+              aria-label="cardnumber"
+              type="text"
+              className="field__input"
+              onClick={() => setSide('front')}
+            />
+            {errors.cardnumber && touched.cardnumber ? (
+              <span className="field__error">{errors.cardnumber}</span>
+            ) : null}
+          </div>
+          <div className="field">
+            <label htmlFor="expirationdate" className="field__label">
+              Expiration (mm/yy)
+            </label>
+            <div className="field__container">
+              <div className="field__container__content">
+                <Field
+                  as="select"
+                  name="expireMM"
+                  aria-label="expireMM"
+                  className="field__select"
+                  onClick={() => setSide('front')}
+                >
+                  <option value="">Month</option>
```

Comment:
It considered as better practice to use map method for generation of multiple items